mlflow/pyfunc/model.py

```diff
@@ -228,17 +229,18 @@ def _save_model_with_class_artifacts_params(
     if artifacts:
         saved_artifacts_config = {}
         with TempDir() as tmp_artifacts_dir:
-            tmp_artifacts_config = {}
             saved_artifacts_dir_subpath = "artifacts"
             for artifact_name, artifact_uri in artifacts.items():
-                tmp_artifact_path = _download_artifact_from_uri(
-                    artifact_uri=artifact_uri, output_path=tmp_artifacts_dir.path()
-                )
-                tmp_artifacts_config[artifact_name] = tmp_artifact_path
-                saved_artifact_subpath = posixpath.join(
-                    saved_artifacts_dir_subpath,
-                    os.path.relpath(path=tmp_artifact_path, start=tmp_artifacts_dir.path()),
-                )
+                if artifact_name == "snapshot":
+                    saved_artifact_subpath = artifact_uri
```

Comment:
if `artifact_name == "snapshot"`, we skip the download?

Reply:
Yes, we just point it to the location